meilisearch-http/src/index/search.rs

```diff
@@ -203,6 +218,73 @@ impl Index {
     }
 }
 
+fn parse_formatted_options(
+    attr_to_highlight: &HashSet<String>,
+    attr_to_crop: &[String],
+    query_crop_length: usize,
+    fields_ids_map: &FieldsIdsMap,
+    displayed_ids: &HashSet<u8>,
+    ) -> HashMap<FieldId, FormatOptions> {
+
+    let mut formatted_options = HashMap::new();
+
+    for attr in attr_to_highlight {
+        let new_format = FormatOptions {
+            highlight: true,
+            crop: None,
+        };
+
+        if attr == "*" {
+            for id in displayed_ids {
+                formatted_options.insert(*id, new_format);
+            }
+            break;
+        }
+
+        if let Some(id) = fields_ids_map.id(&attr) {
+            formatted_options.insert(id, new_format);
+        }
+    }
+
+    for attr in attr_to_crop {
+        let mut attr_name = attr.clone();
+        let mut attr_len = query_crop_length;
+
+        let mut split = attr_name.rsplitn(2, ':');
+        attr_name = match split.next().zip(split.next()) {
+            Some((len, name)) => {
+                attr_len = len.parse().unwrap_or(query_crop_length);
+                name.to_string()
+            },
+            None => attr_name,
+        };
+
+        if attr_name == "*" {
+            for id in displayed_ids {
+                formatted_options
+                    .entry(*id)
+                    .and_modify(|f| f.crop = Some(attr_len))
+                    .or_insert(FormatOptions {
+                        highlight: false,
+                        crop: Some(attr_len),
+                    });
+            }
+        }
+
+        if let Some(id) = fields_ids_map.id(&attr_name) {
+            formatted_options
+                .entry(id)
+                .and_modify(|f| f.crop = Some(attr_len))
+                .or_insert(FormatOptions {
+                    highlight: false,
+                    crop: Some(attr_len),
+                });
+        }
```

Comment:
I feel "le cul entre deux chaises" because Kero and Many think the regexp is harder to read 😂 I agree with the readability argument. And the regexp is not more concise: 8 lines against 10 for the regex.
Plus the issue with your implementation is `title:10q` does not return `title` -> my implementation takes `title` and uses the default crop length. This is iso with v0.20.0. Same for `title:`, I need to return `title` and use the default `cropLength`.